cylc/flow/rundb.py

```diff
@@ -1077,3 +1077,19 @@ def select_jobs_for_datastore(
     def vacuum(self):
         """Vacuum to the database."""
         return self.connect().execute("VACUUM")
+
+    def messages_in_db(self, submit, message, timestamp, itask):
+        name = itask.tokens['task']
+        cycle = itask.tokens['cycle']
+        stmt = (
+            r'SELECT * FROM task_events WHERE'
+            fr' name="{name}"'
+            fr' AND cycle="{cycle}"'
+            fr' AND message="{message}"'
+            fr' AND submit_num="{submit}"'
+            fr' AND time="{timestamp}"'
+        )
+        conn = self.connect()
+        previous_message = any(i for i in conn.execute(stmt))
+        conn.close()
```

Comment:
I'm not sure we should close the DB here?

Follow the pattern of the "select_*" methods in this file.